### PR TITLE
[WFLY-17860]: Convert ServerAdd.InstallServerHandler to CapabilityServiceTarget.

### PR DESCRIPTION
**Pull request by ehsavoie**
* Using the CapabilityServiceTarget API.

Jira: https://issues.redhat.com/browse/WFLY-17860

### CODE REVIEW

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ServerAdd.java [hunk 1]
```diff
@@ -316,14 +319,14 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
 
             // Add the ActiveMQ Service
             ServiceName activeMQServiceName = MessagingServices.getActiveMQServiceName(serverName);
-            final ServiceBuilder serviceBuilder = serviceTarget.addService(activeMQServiceName);
-            Supplier pathManager = serviceBuilder.requires(context.getCapabilityServiceName(PATH_MANAGER_CAPABILITY, PathManager.class));
+            final CapabilityServiceBuilder serviceBuilder = capabilityServiceTarget.addCapability(
+                    RuntimeCapability.Builder.of(activeMQServiceName.getCanonicalName(), false, ActiveMQServer.class).build());
```
[the remark this answers is not in the document]
Comment (ehsavoie) [reply messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ServerAdd.java:323]: It wasn't working and I was getting missing service exceptions

[the remark this answers is not in the document]
Comment (ehsavoie) [reply messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ServerAdd.java:323]: no i didn't

[the remark this answers is not in the document]
Comment (bstansberry) [reply messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ServerAdd.java:323]: See also https://wildfly.zulipchat.com/#narrow/stream/210169-jboss-msc/topic/Undeprecate.20addAliases.3F

File: messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQServerService.java [hunk 2]
```diff
@@ -76,7 +76,7 @@
  * @author scott.stark@jboss.org
  * @author Emanuel Muckenhuber
  */
-class ActiveMQServerService implements Service<ActiveMQServer> {
+class ActiveMQServerService implements Service<ActiveMQServer>, Supplier<ActiveMQServer> {
```
Comment (bstansberry) [inline messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQServerService.java:79]: Why implement Supplier? I don't see any use.

File: messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ServerAdd.java [hunk 3]
```diff
@@ -316,14 +319,14 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
 
             // Add the ActiveMQ Service
             ServiceName activeMQServiceName = MessagingServices.getActiveMQServiceName(serverName);
-            final ServiceBuilder serviceBuilder = serviceTarget.addService(activeMQServiceName);
-            Supplier pathManager = serviceBuilder.requires(context.getCapabilityServiceName(PATH_MANAGER_CAPABILITY, PathManager.class));
+            final CapabilityServiceBuilder serviceBuilder = capabilityServiceTarget.addCapability(ACTIVEMQ_SERVER_CAPABILITY);
+            Consumer<ActiveMQServer> consumer = serviceBuilder.provides(activeMQServiceName);
```
Comment (bstansberry) [inline messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ServerAdd.java:323]: Pass ACTIVEMQ_SERVER_CAPABILITY to provides as well. That way the consumer is used for injection into dependents for both service names.

File: messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQServerService.java (hunk 2 above)
Comment (bstansberry) [reply messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQServerService.java:79]: Also, change this to org.jboss.msc.Service and drop the getValue() method.

Comment (bstansberry) [reply messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQServerService.java:79]: Never mind this last part; I forgot about the ActiveMQActivationService.isActiveMQServerActive thing.

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @ehsavoie!

**Pull request merged**